GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,246 @@
+                }
+            }
+
+            this.GetPathsFromGitIndex(enlistment, out List<string> gitFilePaths, out List<string> gitFolderPaths, out List<string> skipWorkTreeFiles);
+
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData pathData = new GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData(
+                gitFolderPaths,
+                gitFilePaths,
+                placeholderFolderPathList,
+                placeholderFilePathList,
+                modifiedPathsFolderList,
+                modifiedPathsFileList,
+                skipWorkTreeFiles);
+            GVFSEnlistmentHealthCalculator enlistmentHealthCalculator = new GVFSEnlistmentHealthCalculator(pathData);
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData = enlistmentHealthCalculator.CalculateStatistics(this.Directory, this.DirectoryDisplayCount);
+
+            this.PrintOutput(enlistment, enlistmentHealthData);
+        }
+
+        private void PrintOutput(
+            GVFSEnlistment enlistment,
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData)
+        {
+            string trackedFilesCountFormatted = enlistmentHealthData.GitTrackedItemsCount.ToString("N0");
+            string placeholderCountFormatted = enlistmentHealthData.PlaceholderCount.ToString("N0");
+            string modifiedPathsCountFormatted = enlistmentHealthData.ModifiedPathsCount.ToString("N0");
+
+            // Calculate spacing for the numbers of total files
+            int longest = Math.Max(trackedFilesCountFormatted.Length, placeholderCountFormatted.Length);
+            longest = Math.Max(longest, modifiedPathsCountFormatted.Length);
+
+            // Sort the dictionary to find the most hydrated directories by percentage
+            List<ValueTuple<string, decimal>> topLevelDirectoriesByHydration = enlistmentHealthData.DirectoryHydrationLevels;
+
+            this.Output.WriteLine("\nRepository health");
+            this.Output.WriteLine("Total files in HEAD commit:           " + trackedFilesCountFormatted.PadLeft(longest) + " | 100%");
+            this.Output.WriteLine("Files managed by VFS for Git (fast):  " + placeholderCountFormatted.PadLeft(longest) + " | " + this.FormatPercent(enlistmentHealthData.PlaceholderPercentage));
+            this.Output.WriteLine("Files managed by git (slow):          " + modifiedPathsCountFormatted.PadLeft(longest) + " | " + this.FormatPercent(enlistmentHealthData.ModifiedPathsPercentage));
+
+            this.Output.WriteLine("\nTotal hydration percentage:           " + this.FormatPercent(enlistmentHealthData.PlaceholderPercentage + enlistmentHealthData.ModifiedPathsPercentage).PadLeft(longest + 7));
+
+            this.Output.WriteLine("\nMost hydrated top level directories:");
+
+            int maxDirectoryNameLength = 0;
+            foreach ((string, decimal) pair in topLevelDirectoriesByHydration)

```

Comment:
This loop should only be over the set of results you are printing, so it should have a `Take()` as in the earlier case.

However, you should probably just use a `Take()` when constructing the list in the first place, as there is no need to pass around the full list forever.